shared/utils/prismic.js

```diff
@@ -14,67 +15,74 @@ function linkResolver(doc) {
     case 'articles':
       return '/articles';
     case 'article':
-      return `/articles/${doc.slug || doc.uid}`;
+      return `/articles/${doc.uid}`;
     case 'custom_page':
-      return `/${doc.slug || doc.uid}`;
+      return `/${doc.uid}`;
     default:
       return '/';
   }
 }
 
-function renderAsText(field) {
-  try {
-    return asText(field);
-  } catch (e) {
-    console.warn('unable to render field as text', field);
-  }
+const SHOULD_WARN = true;
 
-  return '';
+function warn(...m) {
+  if (SHOULD_WARN) {
+    console.warn(...m);
+  }
 }
 
-function renderAsComponentTree(field) {
-  if (!field) {
-    console.warn('cannot render undefined field');
-    return null;
+function get(obj, path = undefined, defaultValue = '') {
+  const value = path ? _get(obj, path) : obj;
+
+  if (!value) {
+    return defaultValue;
+  }
+
+  if (typeof value === 'string') {
+    return value;
   }
 
   try {
-    const result = RichText.render(field, linkResolver);
-    return result;
+    return asText(value).trim();
   } catch (e) {
-    console.warn('unable to render field as component tree', field);
+    warn(`unable to render field "${path}" as text`, e);
   }
-  return null;
+
+  return defaultValue;
 }
 
-function renderDefault(field) {
-  if (field && typeof field === 'object') {
-    const keys = Object.keys(field);
-    if (keys.length === 1 && keys[0] === 'link_type') {
-      return null;
-    }
-  }
+function getCollection(obj, path) {
+  const value = path ? _get(obj, path) : obj;
 
-  return field;
+  return Array.isArray(value) ? value : [];
 }
 
-function getField(field, type = '') {
-  switch (type) {
-    case 'title':
-      return renderAsText(field);
-    case 'text':
-      return renderAsText(field);
-    case 'richtext':
-      return renderAsComponentTree(field);
-    case 'body':
-      // slices
-      return field || [];
-    default:
-      return renderDefault(field);
+function getObject(obj, path) {
+  const value = path ? _get(obj, path) : obj;
+
+  return !(typeof obj !== 'object' || Array.isArray(obj)) ? value : {};
```

Comment:
`typeof null` is 'object' as well. Maybe use `isObject`?

Also this line is awkward to read and too terse for me.

Reply:
Yes, `_isPlainObject` does the trick since `isObject` also returns `true` for array, func etc.

It's also checking the wrong variable 🙈